base/compiler/abstractinterpretation.jl

```diff
@@ -350,7 +350,7 @@ function precise_container_type(@nospecialize(arg), @nospecialize(typ), vtypes::
         else
             return Any[ p for p in tti0.parameters ]
         end
-    elseif tti0 <: Array
+    elseif tti0 isa Type{<:Array{T}} where T
```

Comment:
?

aren't these supposed to be equivalent, with the subtyping test likely being more reliable